foyer/tests/test_gaff.py

```diff
@@ -0,0 +1,81 @@
+import glob
+import itertools as it
+import os
+
+import parmed as pmd
+from pkg_resources import resource_filename
+import pytest
+
+from foyer import Forcefield
+from foyer.tests.utils import atomtype
+
+GAFF = Forcefield(name='gaff')
+
+GAFF_TESTFILES_DIR = resource_filename('foyer', 'gaff_validation')
+
+class TestGAFF(object):
+
+    @pytest.fixture(autouse=True)
+    def initdir(self, tmpdir):
+        tmpdir.chdir()
+
+    top_files = glob.glob(os.path.join(GAFF_TESTFILES_DIR, '*/*.top'))
+    mol2_files = glob.glob(os.path.join(GAFF_TESTFILES_DIR, '*/*.mol2'))
+
+    # Please update this file if you implement atom typing for a test case.
+    # You can automatically update the files by running the below function
+    # `find_correctly_implemented`.
+    implemented_tests_path = os.path.join(os.path.dirname(__file__),
+                                          'implemented_gaff_tests.txt')
+    with open(implemented_tests_path) as f:
+        correctly_implemented = [line.strip() for line in f]
+
+    def find_correctly_implemented(self):
+        with open(self.implemented_tests_path, 'a') as fh:
+            for mol_path in it.chain(self.top_files, self.mol2_files):
+                _, mol_file = os.path.split(mol_path)
+                mol_name, ext = os.path.splitext(mol_file)
+                try:
+                   self.test_atomtyping(mol_name)
+                except Exception as e:
+                    print(e)
+                    continue
+                else:
+                    if mol_name not in self.correctly_implemented:
+                        fh.write('{}\n'.format(mol_name))
+
+    @pytest.mark.parametrize('mol_name', correctly_implemented)
+    def test_atomtyping(self, mol_name, testfiles_dir=GAFF_TESTFILES_DIR):
+        files = glob.glob(os.path.join(testfiles_dir, mol_name, '*'))
+        for mol_file in files:
+            _, ext = os.path.splitext(mol_file)
+            if ext == '.top':
+                top_filename = '{}.top'.format(mol_name)
+                pdb_filename = '{}.pdb'.format(mol_name)
+                top_path = os.path.join(testfiles_dir, mol_name, top_filename)
+                pdb_path = os.path.join(testfiles_dir, mol_name, pdb_filename)
+                structure = pmd.load_file(top_path, xyz=pdb_path, parametrize=False)
+            elif ext == '.mol2':
+                mol2_path = os.path.join(testfiles_dir, mol_name, mol_file)
+                structure = pmd.load_file(mol2_path, structure=True)
+        atomtype(structure, GAFF)
+#
+#    def test_full_parametrization(self):
```

Comment:
Might be a good idea to uncomment and update this test just to make sure the parametrization is working properly, since for the GAFF tests you use a PDB structure file instead of a GRO structure file.